### ISSUE TITLE
rcfiles: We need to cope correctly when the rcfile does not have a newline at the end and we need to append the source command.

### ISSUE THREAD

**Issue by roxennnn** [rust-lang/rustup#2636]
I am totally new to Rust and I wanted to install it. I followed the installation guide on the [website](https://www.rust-lang.org/learn/get-started). I use Linux Mint, thus I ran the following in a terminal:

```bash
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```
After this, when I open a new terminal, the ``bash: export: `/home/user/.cargo/env': not a valid identifier`` message appears on top of the terminal.

I figured out it is due to the following line which has been added during the installation in the `.bashrc` file:

```bash
export PATH=$PATH:$HOME/.local/binsource "$HOME/.cargo/env"
```

In the end, I solved the identifier error changing the previous line in the `.bashrc` file with this:

```bash
export PATH=$PATH:$HOME/.local/binsource:$HOME/.cargo/env
```

I open this issue because I think fixing this in the installation procedure could help some new inexperienced user (like me) facing it at the end of the installation procedure

**Comment by emilyalbini** [comment]
Transferred the issue!

**Comment by kinnison** [comment]
Thank you Pietro.  It looks like @roxennnn  had a `.bashrc` whose last line lacked a newline at the end.

I think it's definitely a bug that we don't cope gracefully with this and so I'll add this to the 1.24.0 milestone to fix.

**Comment by azav** [comment]
This is not closed.  

I just had the exact same issue on Big Sur after running the indicated curl script from https://rustup.rs

curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh

Last login: Sat Apr 10 19:07:26 on ttys018
-bash: export: `/Users/zav/.cargo/env': not a valid identifier

FYI, I've now run the installer 3 times.

Finally, this line below is written into my .bash_profile and I no longer have the error.  

source "$HOME/.cargo/env"

**_Is an older version of the installer installed on one of your load balancers?_**

Platform: MacOS Big Sur, 11.2, M1 Mac Mini.

**Comment by kinnison** [comment]
@azav Unfortunately there's no way in a PR to label an issue rather than close it.  The bug is *fixed* it's just not *released* yet.

This will be fixed in 1.24 which we hope to release fairly soon.
